debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -46,38 +46,27 @@
  * {@link #KEY_FIELD_REPLACEMENT} may be used to change it. For instance, in our above example, we might choose to
  * make the identifier `db_shard1` and `db_shard2` respectively.
  *
- * @author David Leibovic
  * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author David Leibovic
+ * @author Mario Mueller
  */
 public class ByLogicalTableRouter<R extends ConnectRecord<R>> implements Transformation<R> {
 
-    private final Logger logger = LoggerFactory.getLogger(getClass());
-    private Pattern topicRegex;
-    private String topicReplacement;
-    private Pattern keyFieldRegex;
-    private String keyFieldReplacement;
-    private String keyFieldName;
-    private Cache<Schema, Schema> keySchemaUpdateCache;
-    private Cache<Schema, Schema> envelopeSchemaUpdateCache;
-    private final AvroValidator schemaNameValidator = AvroValidator.create(logger);
-
     private static final Field TOPIC_REGEX = Field.create("topic.regex")
             .withDisplayName("Topic regex")
             .withType(ConfigDef.Type.STRING)
             .withWidth(ConfigDef.Width.LONG)
             .withImportance(ConfigDef.Importance.LOW)
             .withValidation(Field::isRequired, Field::isRegex)
             .withDescription("The regex used for extracting the name of the logical table from the original topic name.");
-
     private static final Field TOPIC_REPLACEMENT = Field.create("topic.replacement")
             .withDisplayName("Topic replacement")
             .withType(ConfigDef.Type.STRING)
             .withWidth(ConfigDef.Width.LONG)
             .withImportance(ConfigDef.Importance.LOW)
-            .withValidation(Field::isRequired)
+            .withValidation(Field::isNotNull)
```

Comment:
Just recognised my own confusion here. Thanks for catching this. It is required, as the support for the use of back-references in the regex removes the need of an empty string. Just match what you want and make the replacement `$1`.